maestro-ios/src/main/java/ios/simctl/SimctlIOSDevice.kt

```diff
@@ -142,6 +142,24 @@ class SimctlIOSDevice(
             .also { screenRecording = null }
     }
 
+    override fun addMedia(path: String) {
+        backupSimulatorMedia()
+        LocalSimulatorUtils.addMedia(deviceId, path)
+    }
+
+    private fun backupSimulatorMedia() {
+        if (!LocalSimulatorUtils.photoDataBackupExists()) {
```

Comment:
does it make sense to make a backup on every new addMedia call, not only when lazy var is not initialized?

Reply:
I am doing it once before we add media because we want the data to come back to the original state where it was before adding media, for the simulator. 

If we do it for each call then it would keep the recent photos in case of addition of multiple media files.